### DISCUSSION TITLE
Testing in Dev Environment - How to Test Locally

### DISCUSSION THREAD

**Discussion by dlamprey** [iamvishnusankar/next-sitemap#658] [Q&A]
This package looks good, but I would like to see it work locally before promoting to any other environment. Is there a way I can do that?

**Comment by vzrenggamani** [comment]
1. You do build the project first.
2. Run the local server like `npm run start`
3. Access the sitemap from `localhost:3000/sitemap.xml`

or just see the `public` folder every run a build

**Reply by dlamprey** [reply]
thanks @vzrenggamani for your reply. that didn't work for me. i did this:

1. npm i next-sitemap
2. create next-sitemap.config.js (in root with sample code provided in read me)
3. npm run build (clean, warnings only)
4. npm run start
5. localhost:3000/sitemap.xml -> 404 error (and no errors in server-side console)

the read me indicates the need for a post build script. do i need that ror a local build? if so, how do i do that?

> Read me: Add next-sitemap as your postbuild script

**Reply by vzrenggamani** [reply]
Yeah @dlamprey, you are still need to add the `postbuild` into your package.json. just add these line to your package.json script 
```js
"postbuild": "next-sitemap"
```

and try to run build and start the server

**Comment by dlamprey** [comment]
That worked! I would add that detail to your read.me to help others like me.

Another question... I would like to have a different frequency value used for certain pages. It looks like I could use the transform option to do this, but am having trouble understanding how it works.

the documentation has this:

`if (customLimitedField(path)) {`

but i cannot find customLimitedField in your source code anywhere other than in the documentation. Is that just an example of a function that could be created to return a boolean value? should that custom function be added after this?

`module.exports = {...}`

**Reply by vzrenggamani** [reply]
Glad to @dlamprey, regarding these function, that are custom function that you need to define yourself. That just an example how to use custom value based on value returned from the function. Hope this helps you..

**Comment by dlamprey** [comment]
thanks for the clarification - that would also be something else that you could note in your documentation. i was able to get your package to work the way i wanted, though i ran into an issue with the transform function - i wasn't able to import a js file in the next-sitemap.config.js file - i had to put the functions directly in the file.

I suggest changing the structure of the next-sitemap.config.js file such that imports are supported unless you know of another way I could do that.

You could replace this:

`module.exports = {...}`

with this:

`const module.exports = {`

and then have your code could import the config file and imports could be used within it.

**Reply by vzrenggamani** [reply]
Glad to hear that @dlamprey, Its better to keep the custom function inside the `next-sitemap.config.js`, its very convenient way to keep things not break / easier troubleshooting.

If you wanted to, you could send pull request to this repository regarding issues / feature request that you wanted to appear :)

if you don't mind, could you mark one of the conversation as "Best Answer" to help others see which solutions are worked for this case. It will be huge help for someone bumping to this same issues 💯 

Have a nice day
